openrank/apiserver/models.py

```diff
@@ -1,3 +1,126 @@
 from django.db import models
 
 # Create your models here.
+#All the Comments and doc-strings may contain Opinionated Views.
+#These are not the final docs. Constructive Negation is welcome!  ex  you may find the comments too-verbose
+
+class Contest(models.Model):
+    name=models.CharField()
+    description=models.TextField()
+    url = models.URLField()
+    is_public = models.BooleanField()
+    starttime = models.TimeField()
+    user_id = models.ForeignKey(User)
+    position = models.ForeignKey(Position)
+    cutoff_duration = models.IntegerField()
+    login_start_time = models.TimeField()
+    password = models.charField(max_length=45)
+
+
+class Position(models.models):
+	job_title = models.charField()
+
+
+
+
+
+class User(models.Model):
+	'''Currently these fiels are planned
+	1)name -eh just the name.     should there be a max length? for name? 
+	2)email - email of the contestant.  BuiltIn  EmailField from django is being used.
+	3)password-
+	4)phone_number- an IntegerField.    how will we validate this? is there any Buildin  (like how it is for Email)
+
+'''
+
+    role = models.OneToOneField()  # A Role can oprionally be assigned to a User. But a User MUST be assigned to SOME ROLE
+	name  = models.CharField(max_length=60)
+	email = models.EmailField(dmax_length=60)
+	password = models.CharField(max_length=50)
+	phone_number = models.charField(max_length=10)
```

Comment:
increase the length so that it supports international numbers and extensions as well.

Reply:
I think it's better if we stick to Indian numbers and also add min_length=10

Reply:
why is it better to to stick to indian numbers, aren't we developing this with the intention of catering to many organisations? what do we gain by setting such limitations or what is the cost of not setting that limitation? Also, even in indian numbers, if we want to support extensions numbers etc., then 10 will not be sufficient. even without that STD code of vizag is 0891 and phone numbers will contain 7 digits there it wilbe 11 digits also our model should always be flexible for future expansion, we may choose to restrict in UI if we want to restrict.

Reply:
OK, we will remove that restriction and change max_length to some higher number